src/electionguard/decryption/coordinator.c

```diff
@@ -70,7 +70,15 @@ Decryption_Coordinator_new(uint32_t num_trustees, uint32_t threshold)
     return result;
 }
 
-void Decryption_Coordinator_free(Decryption_Coordinator c) { free(c); }
+void Decryption_Coordinator_free(Decryption_Coordinator c) 
+{
+    for(uint32_t i = 0; i < c->num_tallies; i++)
+    {
+        Crypto_encryption_rep_free(&c->tallies[i]);
```

Comment:
The code is assuming that c->tallies[i] is either null or properly allocated. Is this true?

Reply:
it is explicitly set `Decryption_Coordinator_receive_share` however I added a check for `tallies_initialized` to ensure we don't enter the loop without that function being run